draft-irtf-cfrg-hpke.md

```diff
@@ -375,133 +375,126 @@ and {{encryption-context}} for details.
+  - `Np` : length in bytes of a point encoded as a byte string
 
 * Arithmetic Operators:
   - `+` : point/point addition : `G x G -> G`
   - `.` : scalar/point multiplication : `F x G -> G`
   - `++` : scalar/scalar addition : `F x F -> F`
   - `**` : scalar/scalar multiplication : `F x F -> F`
-  - `||` : string concatenation
 
 * Type Conversion Algorithms:
-  - `int_to_string(a_c, len)` : converts scalar `a` to string of `len` octets
-  - `string_to_int(a_s)` : converts an octet string `a_string` to scalar
-  - `point_to_string(a)` : converts point to string of `ptLen` octets
-  - `string_to_point(a_s)` : converts string of `ptLen` octets to point 
-    Returns INVALID if the octet string does not convert to a valid point
+  - `SerializePoint(a)` : converts a point `a` to string of `Np` bytes
+  - `DeserializePoint(str)` : converts a string `str` of length `Np`
+    bytes to a point. (note: this function can raise an error if the
+    byte string does not convert to a valid point)
 
 * Functions:
-  - `Hash(x)`: cryptographic hash outputing strings of `hLen` octets
-  - `sk_gen()`: returns fresh uniform random secret key scalar
-  - `pk(sk)`: maps secret key `sk` to its public key
-  - `DeriveKeyPair(ikm)`: deterministically derives key pair from ikm. (Implemented 
-    in section {{derive-key-pair}}.)
-
-Then, assuming that finding the discrete log of uniform random elements of `G` is
-computationally intractible, we can  implement the KEM related cryptographic dependencies
-of HPKE from the KDF and the interface to `G` as follows. The resulting KEM is called 
-`DHKEM(Group, KDF)` where `Group` denotes the Discrete Log group and `KDF` the KDF. 
+  - `sk_gen()`: returns a fresh uniform random secret key scalar
+  - `DeriveKeyPair(ikm)`: deterministically derives a key pair from `ikm`,
+    as described in {{derive-key-pair}}.
+  - `split(str, L)`: splits the byte string `str` into two substrings,
+    the first one consisting of the `L` first bytes of `str`, and the
+    second one of the remaining bytes of `str`.
+
+Then, assuming that finding the discrete log of uniformly random elements of `G` is
+computationally intractible, we can  implement the KEM-related cryptographic dependencies
+of HPKE from the KDF and the interface to `G` as follows. The resulting KEM is called
+`DHKEM(Group, KDF)` where `Group` denotes the Discrete Log group and `KDF` the KDF.
 The function parameters `pkR` and `pkS` are deserialized public keys (i.e. points in `G`).
-{{derive-key-pair}} contains the `DeriveKeyPair` function specification for DHKEMs 
+{{derive-key-pair}} contains the `DeriveKeyPair` function specification for DHKEMs
 defined in this document.
 
-* Constants used internally by DHKEM:
-  - `zero_string = 0x00 = int_to_string(0,1)` : an octet with value `0`
-  - `encap_string = 0x10 = int_to_string(8,1)` : an octet with value `8`
-  - `authencap_string = 0x11 = int_to_string(9,1)` : an octet with value `9`
```

Comment:
Why did you choose to include the bytes `encap_string` and `authencap_string` to distinguish between Encap and AuthEncap? I would say, as the length of their ikm is different (AuthEncap has one serialized point more), that already separates the input domains of the hash functions enough to justify that the outputs of the hash function calls in Encap are independent of the ones in AuthEncap.